WoodStar/WoodStar/TdsChannel.cs

```diff
@@ -0,0 +1,156 @@
+        {
+            _pipeReader = PipeReader.Create(stream);
+            _stream = stream;
+            _sslStream = null!;
+            _sslPipeReader = null!;
+        }
+
+        public void SetSslStream(SslStream sslStream)
+        {
+            _sslStream = sslStream;
+            _sslPipeReader = PipeReader.Create(sslStream);
+        }
+
+        public async Task Prelogin()
+        {
+            var preloginStream = new PreloginStream(0, 0, 1, 0, EncryptionOption.Off, null, 0, false, null, null, null);
+            var header = new TdsHeader(PacketType.PreLogin, PacketStatus.EOM, preloginStream.Length + TdsHeader.HeaderSize, 0, 1);
+            var preloginBuffer = ArrayPool<byte>.Shared.Rent(header.Length);
+            header.WriteToBuffer(preloginBuffer);
+            preloginStream.WriteToBuffer(preloginBuffer);
+
+            await _stream.WriteAsync(preloginBuffer, 0, header.Length);
+            await _stream.FlushAsync();
+
+            ArrayPool<byte>.Shared.Return(preloginBuffer);
+
+            var preloginResponse = await ReadNextPacketAsync(ResponseType.Prelogin);
+            if (preloginResponse is PreloginPacket preloginPacket)
+            {
+                // Assign version etc info to connection
+            }
+        }
+
+        public async Task Login(string username, string password, string database)
+        {
+            var login7Stream = new Login7Stream
+            {
+                UserName = username,
+                Password = password,
+                Database = database,
+            };
+            var header = new TdsHeader(PacketType.Login, PacketStatus.EOM, login7Stream.Length + TdsHeader.HeaderSize, 0, 1);
+            var loginBuffer = ArrayPool<byte>.Shared.Rent(header.Length);
+            header.WriteToBuffer(loginBuffer);
+            login7Stream.WriteToBuffer(loginBuffer);
+
+            await _sslStream.WriteAsync(loginBuffer, 0, header.Length);
+            await _sslStream.FlushAsync();
+
+            ArrayPool<byte>.Shared.Return(loginBuffer);
+
+            await ReadNextPacketAsync(ResponseType.Login);
+        }
+
+        public async Task ExecuteSqlAsync(string sql)
+        {
+            var sqlBatchStream = new SqlBatchStream(
+                new StreamHeader
+                {
+                    TransactionDescriptor = new TransactionDescriptorHeader(0, 1)
+                },
+                sql);
+
+            var header = new TdsHeader(PacketType.SQLBatch, PacketStatus.EOM, sqlBatchStream.Length + TdsHeader.HeaderSize, 0, 1);
+            var sqlBatchBuffer = ArrayPool<byte>.Shared.Rent(header.Length);
+            header.WriteToBuffer(sqlBatchBuffer);
+            sqlBatchStream.WriteToBuffer(sqlBatchBuffer);
+            var b = string.Join(",", sqlBatchBuffer);
+            var c = string.Join(" ", sqlBatchBuffer.Select(e => e.ToString("X2")));
+
+            await _sslStream.WriteAsync(sqlBatchBuffer, 0, header.Length);
```

Comment:
Here (and below) the request is supposed to be send over a non-ssl stream. Essentially, since `EncryptionOption.Off` is send, there is no ssl required at all (except for login).